packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart

```diff
@@ -0,0 +1,366 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../../artifacts.dart';
+import '../../base/common.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../convert.dart';
+import '../../devfs.dart';
+import '../build_system.dart';
+import 'dart.dart';
+
+/// The build define controlling whether icon fonts should be stripped down to
+/// only the glyphs used by the application.
+const String kIconTreeShakerFlag = 'TreeShakeIcons';
+
+/// Whether icon font subsetting is enabled by default.
+const bool kIconTreeShakerEnabledDefault = false;
+
+List<Map<String, dynamic>> _getList(dynamic object, String errorMessage) {
+  try {
+    return (object as List<dynamic>).cast<Map<String, dynamic>>();
+  } on CastError catch (_) {
+    throw IconTreeShakerException._(errorMessage);
+  }
+}
+
+/// A class that wraps the functionality of the const finder package and the
+/// font subset utility to tree shake unused icons from fonts.
+class IconTreeShaker {
+  /// Creates a wrapper for icon font subsetting.
+  ///
+  /// The environment parameter must not be null.
+  ///
+  /// If the `fontManifest` parameter is null, [enabled] will return false since
+  /// there are no fonts to shake.
+  ///
+  /// The constructor will validate the environment and print a warning if
+  /// font subsetting has been requested in a debug build mode.
+  IconTreeShaker(
+    this._environment,
+    DevFSStringContent fontManifest, {
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required FileSystem fs,
```

Comment:
nit: name this `fileSystem` and not `fs` to avoid confusion with the global

Reply:
Done